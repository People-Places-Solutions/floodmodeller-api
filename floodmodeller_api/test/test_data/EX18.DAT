
#REVISION#1
        34     0.750     0.900     0.100     0.001        12
    10.000     0.010     0.010     0.700     0.100     0.700     0.000
RAD FILE

END GENERAL
QTBDY
QT1
         3     0.000     0.000     HOURS    EXTEND    LINEAR          
     0.050     0.000
     0.300     5.000
     0.050     9.000
QTBDY
QT2
         3     0.000     0.000     HOURS    EXTEND    LINEAR          
     0.050     0.000
     0.300     5.000
     0.050     9.000
QTBDY
S1d
         3     0.000     0.000     HOURS    EXTEND    LINEAR          
     0.050     0.000
     0.750     2.000
     0.050     9.000
RESERVOIR #revision#1
S0          QT1         QT2

         5
    19.000   800.000
    20.000  1000.000
    21.000  1200.000
    22.000  1400.000
    23.000  1600.000
      0000      0000         0
WEIR
S0          S1u
     1.500
     1.000     1.000     1.500    21.000     0.800
JUNCTION
OPEN
S1u         S1          S1d
RIVER
SECTION
S1
     5.000              0.0001
         5
     0.000    25.000     0.030     1.000                0.00      0.00
     0.100    20.100     0.030     1.000                0.00      0.00
     1.000    20.100     0.030     1.000                0.00      0.00
     2.000    20.100     0.030     1.000                0.00      0.00
     2.100    25.000     0.030     1.000                0.00      0.00
RIVER
SECTION
S2
     0.000              0.0001
         5
     0.000    25.000     0.030     1.000                0.00      0.00
     0.100    20.000     0.030     1.000                0.00      0.00
     1.000    20.000     0.030     1.000                0.00      0.00
     2.000    20.000     0.030     1.000                0.00      0.00
     2.100    25.000     0.030     1.000                0.00      0.00
CULVERT
INLET
S2          C2
    0.5200     0.670    0.0380     0.800     0.600         A
     0.000     0.000     0.000     0.000ZERO          STATIC     0.000
CONDUIT
CIRCULAR
C2
   100.000
MANNING
    20.000     1.000
   0.02000   0.01500
REPLICATE
C2_R1
   100.000     0.200         0         0     0.200
REPLICATE
C2_R2
   100.000     0.200         0         0     0.200
REPLICATE
C2_R3
   100.000     0.200         0         0     0.200
REPLICATE
C2_R4
   100.000     0.200         0         0     0.200
CONDUIT
CIRCULAR
C2m
     0.000
MANNING
    19.200     1.000
   0.02000   0.01500
MANHOLE #revision#1
C2m         C2md        C2mH
      20.5         3       1.7       0.9         0
HTBDY
C2mH
         1             SECONDS    EXTEND    LINEAR
    20.500     0.000
CONDUIT
CIRCULAR
C2md
   100.000
MANNING
    19.200     1.000
   0.02000   0.01500
REPLICATE
C2_R5
   100.000     0.200         0         0     0.200
REPLICATE
C2_R6
   100.000     0.200         0         0     0.200
REPLICATE
C2_R7
   100.000     0.200         0         0     0.200
REPLICATE
C2_R8
   100.000     0.200         0         0     0.200
REPLICATE
C2_R9
   100.000     0.200         0         0     0.200
REPLICATE
C2_R10
   100.000     0.200         0         0     0.200
CONDUIT
CIRCULAR
C2d
     0.000
MANNING
    18.000     1.000
   0.02000   0.01500
WEIR
C2d         S3
     1.500
     1.000     1.000     0.600    18.000     0.800
RIVER
SECTION
S3          S3LS
   100.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    18.000     0.030     1.000                0.00      0.00
     1.000    18.000     0.030     1.000                0.00      0.00
     2.000    18.000     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
RIVER
SECTION
S4
     0.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    17.800     0.030     1.000                0.00      0.00
     1.000    17.800     0.030     1.000                0.00      0.00
     2.000    17.800     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
WEIR
S4          S4_drop
     1.500
     1.000     1.000     2.000    17.900     0.800
JUNCTION
OPEN
S4_drop     P_suc       S5
RIVER
SECTION
S5
   100.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    17.700     0.030     1.000                0.00      0.00
     1.000    17.700     0.030     1.000                0.00      0.00
     2.000    17.700     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
RIVER
SECTION
S6
   100.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    17.700     0.030     1.000                0.00      0.00
     1.000    17.700     0.030     1.000                0.00      0.00
     2.000    17.700     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
RIVER
SECTION
S7
   100.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    17.700     0.030     1.000                0.00      0.00
     1.000    17.700     0.030     1.000                0.00      0.00
     2.000    18.000     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
RIVER
SECTION
S8
     0.000              0.0001
         5
     0.000    20.000     0.030     1.000                0.00      0.00
     0.100    17.600     0.030     1.000                0.00      0.00
     1.000    17.600     0.030     1.000                0.00      0.00
     2.000    17.600     0.030     1.000                0.00      0.00
     2.100    20.000     0.030     1.000                0.00      0.00
WEIR
S8          S9
     1.500
     1.000     1.000     2.000    17.700     0.800
OCPUMP
P_suc       P_del
   730.000    19.000     0.200     0.600
LOGICAL
TRIGGER   MODE
         3HOURS     EXTEND
     0.000    MANUAL   STOPPED
     2.500    MANUAL        ON
     8.000    MANUAL   STOPPED
HEAD      FLOW      EFFICIENCY
         3
    17.600     0.100     0.550
    19.000     0.200     0.600
    20.000     0.300     0.550
RULES
3              0.000HOURS     EXTEND
Rule1
IF(LEVEL(S3).LT.18.3)
THEN PUMP=STOPPED
END
Rule2
IF(LEVEL(S6).GT.21.5)
THEN PUMP=STOPPED
END
Rule3
IF(LEVEL(S3).GT.20.3)
THEN PUMP=ON
END
TIME RULE DATA SET
         2
     0.000ALL
   999.000ALL
HTBDY
P_del
         1
     0.000     0.000
HTBDY
S9
         2               HOURS    EXTEND    LINEAR
    17.700     0.000
    17.700    10.000
SPILL
S3LS        CSO
     1.700     0.900
         2
     0.000    20.000      0.00      0.00
   100.000    20.000      0.00      0.00
RESERVOIR #revision#1
CSO

         5
    19.000   800.000
    20.000  1000.000
    21.000  1200.000
    22.000  1200.000
    23.000  1200.000
      0000      0000         0
INITIAL CONDITIONS
 label   ?      flow     stage froude no  velocity     umode    ustate         z
QT1          y     0.050    21.115     0.000     0.000     0.000     0.000     0.000
QT2          y     0.050    21.115     0.000     0.000     0.000     0.000     0.000
S1d          y     0.050    20.544     0.000     0.000     0.000     0.000     0.000
S0           y     0.100    21.115     0.222     0.629     1.000     0.000    21.000
S1u          y     0.100    20.544     0.000     0.000     0.000     0.000    21.000
S1           y     0.150    20.544     0.005     0.026     0.000     0.000   -79.900
S2           y     0.150    20.544     0.290     0.025     1.000     0.000   -80.000
C2           y     0.150    20.315     0.473     0.708     0.000     0.000    20.000
C2_R1        y     0.150    20.116     0.471     0.705     0.000     0.000    19.800
C2_R2        y     0.150    19.913     0.480     0.715     0.000     0.000    19.600
C2_R3        y     0.150    19.743     0.401     0.629     0.000     0.000    19.400
C2_R4        y     0.150    19.657     0.232     0.429     0.000     0.000    19.200
C2m          y     0.150    19.515     0.473     0.707     0.000     0.000    19.200
C2md         y     0.150    19.515     0.473     0.707     0.000     0.000    19.200
C2mH         y     0.000    20.500     0.000     0.000     0.000     0.000    20.500
C2_R5        y     0.150    19.315     0.473     0.708     0.000     0.000    19.000
C2_R6        y     0.150    19.115     0.472     0.707     0.000     0.000    18.800
C2_R7        y     0.150    18.914     0.476     0.711     0.000     0.000    18.600
C2_R8        y     0.150    18.721     0.455     0.689     0.000     0.000    18.400
C2_R9        y     0.150    18.597     0.303     0.516     0.000     0.000    18.200
C2_R10       y     0.150    18.550     0.162     0.339     0.000     0.000    18.000
C2d          y     0.150    18.251     0.735     0.972     1.000     1.000    18.000
S3           y     0.150    18.031     0.006     0.030     0.000     0.000   -82.000
S3LS         y     0.000 -9999.990     0.000     0.000     0.000     0.000    20.000
S4           y     0.150    18.025     0.005     0.028     1.000     0.000   -82.200
S4_drop      y     0.150    17.843     0.009     0.049     0.000     0.000    17.900
P_suc        y     0.000    17.843     0.000     0.000     1.000     0.000     0.000
S5           y     0.150    17.843     0.005     0.028     0.000     0.000   -82.300
S6           y     0.150    17.838     0.005     0.028     0.000     0.000   -82.300
S7           y     0.150    17.831     0.005     0.029     0.000     0.000   -82.300
S8           y     0.150    17.825     0.005     0.028     1.000     0.000   -82.400
S9           y     0.150    17.700     0.000     0.000     0.000     0.000    17.700
P_del        y     0.000     0.000     0.000     0.000     0.000     0.000     0.000
CSO          y     0.000    19.001     0.000     0.000     0.000     0.000     0.000
GISINFO
QTBDY QT1 0 0 0 0 0
QTBDY QT2 0 0 0 0 0
QTBDY S1d 0 0 0 0 0
RESERVOIR S0 0 0 0 0 0
WEIR S0 0 0 0 0 0
JUNCTION OPEN S1u 0 0 0 0 0
RIVER SECTION S1 0 0 0 0 0
RIVER SECTION S2 0 0 0 0 0
CULVERT INLET S2 0 0 0 0 0
CONDUIT CIRCULAR C2 0 0 0 0 0
REPLICATE C2_R1 0 0 0 0 0
REPLICATE C2_R2 0 0 0 0 0
REPLICATE C2_R3 0 0 0 0 0
REPLICATE C2_R4 0 0 0 0 0
CONDUIT CIRCULAR C2m 0 0 0 0 0
MANHOLE C2m 0 0 0 0 0
HTBDY C2mH 0 0 0 0 0
CONDUIT CIRCULAR C2md 0 0 0 0 0
REPLICATE C2_R5 0 0 0 0 0
REPLICATE C2_R6 0 0 0 0 0
REPLICATE C2_R7 0 0 0 0 0
REPLICATE C2_R8 0 0 0 0 0
REPLICATE C2_R9 0 0 0 0 0
REPLICATE C2_R10 0 0 0 0 0
CONDUIT CIRCULAR C2d 0 0 0 0 0
WEIR C2d 0 0 0 0 0
RIVER SECTION S3 0 0 0 0 0
RIVER SECTION S4 0 0 0 0 0
WEIR S4 0 0 0 0 0
JUNCTION OPEN S4_drop 0 0 0 0 0
RIVER SECTION S5 0 0 0 0 0
RIVER SECTION S6 0 0 0 0 0
RIVER SECTION S7 0 0 0 0 0
RIVER SECTION S8 0 0 0 0 0
WEIR S8 0 0 0 0 0
OCPUMP P_suc 0 0 0 0 0
HTBDY P_del 0 0 0 0 0
HTBDY S9 0 0 0 0 0
SPILL S3LS 0 0 0 0 0
RESERVOIR CSO 0 0 0 0 0
QT1 0 0 0 0 0
QT2 0 0 0 0 0
S1d 0 0 0 0 0
S0 0 0 0 0 0
S1u 0 0 0 0 0
S1 0 0 0 0 0
S2 0 0 0 0 0
C2 0 0 0 0 0
C2_R1 0 0 0 0 0
C2_R2 0 0 0 0 0
C2_R3 0 0 0 0 0
C2_R4 0 0 0 0 0
C2m 0 0 0 0 0
C2md 0 0 0 0 0
C2mH 0 0 0 0 0
C2_R5 0 0 0 0 0
C2_R6 0 0 0 0 0
C2_R7 0 0 0 0 0
C2_R8 0 0 0 0 0
C2_R9 0 0 0 0 0
C2_R10 0 0 0 0 0
C2d 0 0 0 0 0
S3 0 0 0 0 0
S3LS 0 0 0 0 0
S4 0 0 0 0 0
S4_drop 0 0 0 0 0
P_suc 0 0 0 0 0
S5 0 0 0 0 0
S6 0 0 0 0 0
S7 0 0 0 0 0
S8 0 0 0 0 0
S9 0 0 0 0 0
P_del 0 0 0 0 0
CSO 0 0 0 0 0
