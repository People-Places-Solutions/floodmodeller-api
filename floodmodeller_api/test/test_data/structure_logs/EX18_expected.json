{
    "(C2,CONDUIT)": {
        "name": "C2",
        "type": "CONDUIT",
        "subtype": "CIRCULAR",
        "comment": "",
        "conduit_data": {
            "length": 100.0,
            "inlet": 0.6,
            "total_length": 500.0
        },
        "dimensions": {
            "invert": 20.0,
            "diameter": 1.0
        },
        "friction": {
            "friction_eq": "MANNING",
            "friction_set": [
                0.015,
                0.02
            ],
            "all_friction": [
                0.015,
                0.02
            ]
        }
    },
    "(C2_R1,REPLICATE)": {
        "name": "C2_R1",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2m"
        }
    },
    "(C2_R2,REPLICATE)": {
        "name": "C2_R2",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2m"
        }
    },
    "(C2_R3,REPLICATE)": {
        "name": "C2_R3",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2m"
        }
    },
    "(C2_R4,REPLICATE)": {
        "name": "C2_R4",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2m"
        }
    },
    "(C2m,CONDUIT)": {
        "name": "C2m",
        "type": "CONDUIT",
        "subtype": "CIRCULAR",
        "comment": "",
        "conduit_data": {
            "length": 0.0
        },
        "dimensions": {
            "invert": 19.2,
            "diameter": 1.0
        },
        "friction": {
            "friction_eq": "MANNING",
            "friction_set": [
                0.015,
                0.02
            ],
            "all_friction": [
                0.015,
                0.02
            ]
        }
    },
    "(C2md,CONDUIT)": {
        "name": "C2md",
        "type": "CONDUIT",
        "subtype": "CIRCULAR",
        "comment": "",
        "conduit_data": {
            "length": 100.0,
            "total_length": 700.0
        },
        "dimensions": {
            "invert": 19.2,
            "diameter": 1.0
        },
        "friction": {
            "friction_eq": "MANNING",
            "friction_set": [
                0.015,
                0.02
            ],
            "all_friction": [
                0.015,
                0.02
            ]
        }
    },
    "(C2_R5,REPLICATE)": {
        "name": "C2_R5",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2_R6,REPLICATE)": {
        "name": "C2_R6",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2_R7,REPLICATE)": {
        "name": "C2_R7",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2_R8,REPLICATE)": {
        "name": "C2_R8",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2_R9,REPLICATE)": {
        "name": "C2_R9",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2_R10,REPLICATE)": {
        "name": "C2_R10",
        "type": "REPLICATE",
        "subtype": null,
        "comment": "",
        "conduit_data": {
            "length": 100.0
        },
        "dimensions": {
            "bed_level_drop": 0.2,
            "mimic": "C2d"
        }
    },
    "(C2d,CONDUIT)": {
        "name": "C2d",
        "type": "CONDUIT",
        "subtype": "CIRCULAR",
        "comment": "",
        "conduit_data": {
            "length": 0.0
        },
        "dimensions": {
            "invert": 18.0,
            "diameter": 1.0
        },
        "friction": {
            "friction_eq": "MANNING",
            "friction_set": [
                0.015,
                0.02
            ],
            "all_friction": [
                0.015,
                0.02
            ]
        }
    },
    "(S0,WEIR)": {
        "name": "S0",
        "type": "WEIR",
        "subtype": null,
        "comment": "",
        "dimensions": {
            "weir_elevation": 21.0,
            "weir_breadth": 1.5
        }
    },
    "(C2d,WEIR)": {
        "name": "C2d",
        "type": "WEIR",
        "subtype": null,
        "comment": "",
        "dimensions": {
            "weir_elevation": 18.0,
            "weir_breadth": 0.6
        }
    },
    "(S4,WEIR)": {
        "name": "S4",
        "type": "WEIR",
        "subtype": null,
        "comment": "",
        "dimensions": {
            "weir_elevation": 17.9,
            "weir_breadth": 2.0
        }
    },
    "(S8,WEIR)": {
        "name": "S8",
        "type": "WEIR",
        "subtype": null,
        "comment": "",
        "dimensions": {
            "weir_elevation": 17.7,
            "weir_breadth": 2.0
        }
    },
    "(S3LS,SPILL)": {
        "name": "S3LS",
        "type": "SPILL",
        "subtype": null,
        "comment": "",
        "dimensions": {
            "invert": 20.0,
            "width": 100.0,
            "weir_coefficient": 1.7,
            "section_data": {
                "x": [
                    0.0,
                    100.0
                ],
                "y": [
                    20.0,
                    20.0
                ]
            }
        }
    }
}